### DISCUSSION TITLE
Custom made RC control

### DISCUSSION THREAD

**Discussion by not7cd** [EdgeTX/edgetx#3641] [Q&A] [answered]
Hello, I wish to integrate a custom RC panel, and I don't want to destroy a hitsuba radio. I have ExpressLRS TX modules, both in nano and micro versions. And I want to talk with them directly. Don't question why we are using RC for something that looks like a rocket, long story short: technical debt.
![image](https://github.com/EdgeTX/edgetx/assets/4332632/e8f398b8-49bc-4a39-bb69-741c5eb0753b)

What I really need is some way to talk CRFS to the TX module, and send inputs straight to it. How I should approach this? Should I just grab `pulses/crossfire.h`, and adapt it somehow? How input data is processed inside EdgeTX? And how does it get out? I need something simple without much reconfigurability on the fly. 

This looks promising.
https://github.com/EdgeTX/edgetx/blob/main/radio/src/pulses/crossfire.cpp

I don't care for this, but is it required? Or is it useful?
https://github.com/EdgeTX/edgetx/blob/main/radio/src/telemetry/crossfire.cpp

**Comment by relaxibus** [comment]
Hi! I'm also working on a rugged industrial-grade line for EdgeTX. Actually all closed source that should be migrated to EdgeTX: https://likeabird.eu/remote-control-systems/

Do you have interest in a collaboration?

**Reply by not7cd** [reply]
I would be happy to use it, as I have quite a tight schedule. As I understand, you are using a custom build of EdgeTX, not parts of it?

Also, really cool station. Definitely a different use case, but it's all buttons down the road. My design is more motivated by cost saving, as TX modules are 10 cheaper than full RC sets.

**Reply by pfeerick** [reply]
Another possibility for cost-cutting to keep in mind depending on the range needed, is that it's possible to use ELRS receivers as transmitters (since they are transceivers anyway) - you would probably just want one with higher "telemetry" output power so it has a decent transmit range. Although you have the dedicated transmitter module already here, so probably no point going down that rabbit hole.

**Comment by pfeerick** [comment, marked as answer]
You might find a project like this worth repurposing the code from... I'm sure I saw another (will link it if I come across it again) but this is the only one I could find in a quick search... https://github.com/kkbin505/Arduino-Transmitter-for-ELRS ... he said he used an external module and implements all the "radio" analog input reading right in the one Arduino "sketch".

https://github.com/ZZ-Cat/CRSFforArduino might be of interest also.

I'm not saying you couldn't scavenge stuff from ETX, but it sounds like it's way more complex than you want.

**Reply by not7cd** [reply]
This is a nice lead. I will look into this.

**Comment by relaxibus** [comment]
for an easy way to go, I have a ready made ESP32 board with CRSF interface. It's what interfaces the NOVA proprietary boards and software. If you just build a small stick and button interface and output our SERaero serial protocol, then connect it to our Gateway ESP32 board, you're done! Including a Windows app for channel configurations.

**Reply by not7cd** [reply]
@relaxibus, is this open source, and can be implemented in our device in a week? Because that's a time frame I'm working in.

**Reply by relaxibus** [reply]
The SERaero protocol is open source. The ESP32 code is only available in hex format as I still don't got the complete code from my friend yet. The ESP32 carrier board is available. Where are you based?
